docs/aws/audit/redshiftmonitoring/rules/allow_cluster_version_upgrade_remediation.mdx

```diff
@@ -1,6 +1,216 @@
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Allow Version Upgrade" setting under the "Configuration" tab. If this setting is set to "No", then the Redshift cluster is not configured to allow version upgrades.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Redshift clusters: Use the following AWS CLI command to list all the Redshift clusters in your account:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+   ```
+   This command will return a list of all Redshift cluster identifiers.
+
+3. Check if Redshift clusters allow version upgrade: For each cluster identifier returned from the previous step, run the following command to check if the cluster allows version upgrades:
+
+   ```
+   aws redshift describe-clusters --cluster-identifier <ClusterIdentifier> --query 'Clusters[*].[AllowVersionUpgrade]' --output text
+   ```
+   Replace `<ClusterIdentifier>` with the actual cluster identifier. This command will return a boolean value. If it returns `true`, it means the cluster allows version upgrades. If it returns `false`, it means the cluster does not allow version upgrades.
+
+4. Repeat step 3 for all clusters: You need to repeat the third step for all the cluster identifiers returned from the second step. This will help you check if all Redshift clusters in your account allow version upgrades.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if Redshift Clusters allow version upgrade in AWS using Python scripts, you can use the Boto3 library, which allows you to write software that makes use of services like Amazon S3, Amazon EC2, etc. Here are the steps:
+
+1. **Import the Boto3 library in Python:**
+   Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+   ```python
+   pip install boto3
+   ```
+
+2. **Create an AWS session using your credentials:**
+   You need to configure your AWS credentials. You can do this in several ways, but the simplest is to use the AWS CLI. Here is a sample script:
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>